jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerBuilder.java

```diff
@@ -64,28 +70,42 @@ public UnwrittenLayer build() throws IOException {
                   for (Path sourceFileRelativePathComponent : sourceFileRelativePath) {
                     subExtractionPath.append('/').append(sourceFileRelativePathComponent);
                   }
-                  filesystemEntries.add(
-                      new TarArchiveEntry(path.toFile(), subExtractionPath.toString()));
+                  TarArchiveEntry tarArchiveEntry =
+                      new TarArchiveEntry(path.toFile(), subExtractionPath.toString());
+                  stripNonReproducibleElements(tarArchiveEntry);
+                  filesystemEntries.add(tarArchiveEntry);
                 });
         continue;
       }
 
       TarArchiveEntry tarArchiveEntry =
           new TarArchiveEntry(sourceFile.toFile(), extractionPath + "/" + sourceFile.getFileName());
-      tarArchiveEntry.setModTime(0);
+      if (enableReproducibleBuilds) {
+        stripNonReproducibleElements(tarArchiveEntry);
+      }
       filesystemEntries.add(tarArchiveEntry);
     }
 
-    TarStreamBuilder tarStreamBuilder = new TarStreamBuilder();
+    // sort for reproducibility
+    filesystemEntries.sort(Comparator.comparing(TarArchiveEntry::getName));
 
     // Adds all the files to a tar stream.
+    TarStreamBuilder tarStreamBuilder = new TarStreamBuilder();
     for (TarArchiveEntry entry : filesystemEntries) {
       tarStreamBuilder.addEntry(entry);
     }
 
     return new UnwrittenLayer(tarStreamBuilder.toBlob());
   }
 
+  private void stripNonReproducibleElements(TarArchiveEntry tarArchiveEntry) {
```

Comment:
This could be static, or be a part of the common non-static method for making a `TarArchiveEntry` as mentioned in the previous comment.